temporal-sdk/src/main/java/io/temporal/failure/ApplicationFailure.java

```diff
@@ -218,4 +274,43 @@ private static String getMessage(String message, String type, boolean nonRetryab
         + ", nonRetryable="
         + nonRetryable;
   }
+
+  public static boolean isBenignApplicationFailure(@Nullable Throwable t) {
```

Comment:
Easier to justify than the Go equivalent I suppose, but I still think this helper may not be needed and may be more confusing than it's worth

Reply:
Moved to an internal failure utils file, it's convenient to have imo